### ISSUE TITLE
Pydantic model validation fails if field name is used instead of alias with `allow_population_by_field_name = True`

### ISSUE THREAD

**Issue by priitlatt** [python/mypy#14862]
**Bug Report**

Given a Pydantic model declaration for which `allow_population_by_field_name` is set to `True` in [model config](https://docs.pydantic.dev/usage/model_config/), Mypy repors an "Unexpected keyword argument" error when the model is initialized with field name instead of alias. 

**To Reproduce**

```python
import pydantic


class Model(pydantic.BaseModel):
    class Config:
        allow_population_by_field_name = True

    field_name: str = pydantic.Field(..., alias="fieldName")


if __name__ == "__main__":
    assert Model(field_name="name").field_name == "name"
```

**Expected Behavior**

No error is reported as Pydantic model can be created using actual field name.
 

**Actual Behavior**

```python
model.py:12: error: Unexpected keyword argument "field_name" for "Model"; did you mean "fieldName"?  [call-arg]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.1.1
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.2

**Comment by cdce8p** [comment]
Now that mypy officially supports `dataclass_transform`, you'll need to use the `pydantic.mypy` plugin (with pydantic ´>=1.10.6`) for it to work.

https://docs.pydantic.dev/mypy_plugin/

**Comment by priitlatt** [comment]
Plugin works. Thanks.

**Comment by spiralman** [comment]
FYI, this does not work with Pydantic 1.x if you use the non-deprecated model class kwargs for adding the option to your class. You will have to use the deprecated inner-class `Config` option, due to https://github.com/pydantic/pydantic/issues/5858 and won't be fixed until https://github.com/pydantic/pydantic/pull/5982 is back-ported to 1.x.

MyPy version: 1.4.1
Pydantic version: 1.10.12
